test/e2e/upgrade/upgrade.go

```diff
@@ -270,14 +270,28 @@ func clusterUpgrade(f *framework.Framework, c configv1client.Interface, dc dynam
 
 	// this is very long.  We should update the clusteroperator junit to give us a duration.
 	maximumDuration := 150 * time.Minute
-	// if upgrades take longer than this, then we will have a junit marker indicating failure.
-	durationToSoftFailure := 75 * time.Minute
+	baseDurationToSoftFailure := 75 * time.Minute
+	durationToSoftFailure := baseDurationToSoftFailure
 
 	infra, err := c.ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
 	framework.ExpectNoError(err)
 	if infra.Status.PlatformStatus.Type == configv1.AWSPlatformType {
 		// due to https://bugzilla.redhat.com/show_bug.cgi?id=1943804 upgrades take ~12 extra minutes on AWS
-		durationToSoftFailure = 105 * time.Minute
+		// and see commit d69db34a816f3ce8a9ab567621d145c5cd2d257f which notes that some AWS upgrades can
+		// take close to 105 minutes total (75 is base duration, so adding 30 more if it's AWS)
+		durationToSoftFailure = (baseDurationToSoftFailure + 30) * time.Minute
+	} else {
+		// if the cluster is on AWS we've already bumped the timeout enough, but if not we need to check if
+		// the CNI is OVN and increase our timeout for that
```

Comment:
OVN and provider-LB delay sources seem orthogonal.  If that's right, can we use:

```go
if infra.Status.PlatformStatus.Type == configv1.AWSPlatformType {
  // due to https://bugzilla.redhat.com/show_bug.cgi?id=1943804 upgrades take ~12 extra minutes on AWS
  durationToSoftFailure += 20 * time.Minute
}
network, err := c.ConfigV1().Networks().Get(context.Background(), "cluster", metav1.GetOptions{})
framework.ExpectNoError(err)
if network.Status.NetworkType == "OVNKubernetes" {
  durationToSoftFailure += 15 * time.Minute
}
```

or similar?

Reply:
but if it's OVN + AWS we will end up with only +15 when we want +30. I can do the platform check second so that even if it's OVN it will still end up with +30, but that was sort of why I did it this way to start with so I could comment the code about
the AWS increase being good enough for OVN anyway. Let me know what's best though. I'd like to get this in so we can start making small progress to improving that ovn-upgrade job